api/openapi_server/controllers/auth_controller.py

```diff
@@ -214,6 +215,43 @@ def signin():
     }
 
 
+def resend_confirmation_code():
+    '''
+    Resends the registration confirmation code to the specified user (identified by email).
+    '''
+
+    if connexion.request.is_json:
+        body = connexion.request.get_json()
+
+    if "email" not in body:
+        raise AuthError({"message": "email invalid"}, 400)
+
+    secret_hash = get_secret_hash(body['email'])
+
+    try:
+        email = body['email']
+        response = userClient.resend_confirmation_code(
+            ClientId=COGNITO_CLIENT_ID,
+            SecretHash=secret_hash,
+            Username=email,
+        )
+        return response
+    except botocore.exceptions.ClientError as error:
+        match error.response['Error']['Code']:
+            case 'UserNotFoundException':
+                msg = "User not found. Confirmation not sent."
```

Comment:
I really like the more specific breakdown of possible errors. Might be worth doing this with the other routes